discovery_engine_core/web-api/src/routes.rs

```diff
@@ -53,7 +57,14 @@ fn delete_internal_state(db: Db) -> impl Filter<Extract = impl Reply, Error = Re
 
 // PATH /user/:user_id
 fn user_path() -> impl Filter<Extract = (UserId,), Error = Rejection> + Clone {
-    warp::path("user").and(warp::path::param::<UserId>())
+    warp::path("user")
+        .and(warp::path::param::<String>())
+        .and_then(|user_id: String| async move {
+            urlencoding::decode(&user_id)
+                .map_err(Error::UserIdUtf8Conversion)
+                .and_then(UserId::new)
+                .map_err(|_| warp::reject())
```

Comment:
This produces a `404` instead of a `400`.

Reply:
I changed this to return the error from `UserId::new`. That can be handled correctly in the task we have for the errors.